specifyweb/frontend/js_src/lib/components/AppResources/Editor.tsx

```diff
@@ -213,7 +220,11 @@ export function AppResourceEditor({
   );
 
   return typeof resourceData === 'object' ? (
-    <Container.Base className="flex-1 overflow-auto sm:overflow-visible">
+    <Container.Base
+      className={`flex-1 overflow-auto
+        ${lineWrap ? 'sm:overflow-visible' : ''}
```

Comment:
I don't think this has to be conditional based on user preference
Whenever possible, try to have the code path be as common as possible for all users regardless of preferences - that way there are fewer things that could break and fewer things that need testing

So here, try to come up with styles that work both when line wrap is disabled, and when screen is narrow

Reply:
really, the overflow style should be set on the text editor container directly, not the page container (i.e if editor has a long line, and horizontal scrollbar appears, the header should not be scrolling off - header should stay in place, and only the editor should scroll)

you may still have overflow-auto on the page container as a fallback, but setting it on the text editor container is probably a better idea